components/TextFields/examples/TextFieldSemanticColorThemer.swift

```diff
@@ -0,0 +1,259 @@
+
+    NSLayoutConstraint.activate(NSLayoutConstraint.constraints(
+      withVisualFormat: "V:|[scrollView]|",
+      options: [],
+      metrics: nil,
+      views: ["scrollView": scrollView]))
+    NSLayoutConstraint.activate(NSLayoutConstraint.constraints(withVisualFormat: "H:|[scrollView]|",
+                                                               options: [],
+                                                               metrics: nil,
+                                                               views: ["scrollView": scrollView]))
+    let marginOffset: CGFloat = 16
+    let margins = UIEdgeInsets(top: 0, left: marginOffset, bottom: 0, right: marginOffset)
+
+    scrollView.layoutMargins = margins
+  }
+
+
+  func addGestureRecognizer() {
+    let tapRecognizer = UITapGestureRecognizer(target: self,
+                                               action: #selector(tapDidTouch(sender: )))
+    self.scrollView.addGestureRecognizer(tapRecognizer)
+  }
+
+  // MARK: - Actions
+
+  @objc func tapDidTouch(sender: Any) {
+    self.view.endEditing(true)
+  }
+
+}
+
+extension TextFieldSemanticColorThemer: UITextFieldDelegate {
+  func textField(_ textField: UITextField,
+                 shouldChangeCharactersIn range: NSRange,
+                 replacementString string: String) -> Bool {
+    guard let _ = textField.text else {
+      return true
+    }
+
+    return true
+  }
+
+  func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+    return false
+  }
+}
+
+extension TextFieldSemanticColorThemer: UITextViewDelegate {
+  func textViewDidEndEditing(_ textView: UITextView) {
+    print(textView.text)
+  }
+}
+
+// MARK: - Keyboard Handling
+
+extension TextFieldSemanticColorThemer {
+  func registerKeyboardNotifications() {
+    let notificationCenter = NotificationCenter.default
+    notificationCenter.addObserver(
+      self,
+      selector: #selector(keyboardWillShow(notif:)),
+      name: .UIKeyboardWillChangeFrame,
+      object: nil)
+    notificationCenter.addObserver(
+      self,
+      selector: #selector(keyboardWillShow(notif:)),
+      name: .UIKeyboardWillShow,
+      object: nil)
+    notificationCenter.addObserver(
+      self,
+      selector: #selector(keyboardWillHide(notif:)),
+      name: .UIKeyboardWillHide,
+      object: nil)
+  }
+
+  @objc func keyboardWillShow(notif: Notification) {
+    guard let frame = notif.userInfo?[UIKeyboardFrameEndUserInfoKey] as? CGRect else {
+      return
+    }
+    scrollView.contentInset = UIEdgeInsets(top: 0.0,
+                                           left: 0.0,
+                                           bottom: frame.height,
+                                           right: 0.0)
+  }
+
+  @objc func keyboardWillHide(notif: Notification) {
+    scrollView.contentInset = UIEdgeInsets()
+  }
+}
+
+// MARK: - Status Bar Style
+
+extension TextFieldSemanticColorThemer {
+  @objc class func catalogBreadcrumbs() -> [String] {
```

Comment:
if you want this part of the catalog app don't forget to add catalogIsPresentable